core/io/resource_loader.cpp

```diff
@@ -1198,6 +1198,13 @@ bool ResourceLoader::has_custom_uid_support(const String &p_path) {
 
 bool ResourceLoader::should_create_uid_file(const String &p_path) {
 	const String local_path = _validate_local_path(p_path);
+
+	bool generate_uid_files = GLOBAL_GET("editor/resource/generate_uid_files");
+	bool generate_uid_files_for_addons = GLOBAL_GET("editor/resource/generate_uid_files_for_addons");
```

Comment:
these should be per project, not per computer/editor settings
Everyone in the team should share the same setup